plugins/MauticFocusBundle/Entity/StatRepository.php

```diff
@@ -39,4 +42,53 @@ public function getStats($id, $type, $fromDate = null)
 
         return $q->getQuery()->getArrayResult();
     }
+
+    /**
+     * @param array<string, mixed> $options
+     *
+     * @return array<string, mixed>
+     */
+    public function getStatsViewByLead(int $leadId, array $options = []): array
+    {
+        return $this->getStatsByLeadAndType($leadId, Stat::TYPE_NOTIFICATION, $options);
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     *
+     * @return array<string, mixed>
+     */
+    public function getStatsClickByLead(int $leadId, array $options = []): array
+    {
+        return $this->getStatsByLeadAndType($leadId, Stat::TYPE_CLICK, $options);
+    }
+
+    /**
+     * @param array<string, mixed> $options
+     *
+     * @return array<string, mixed>
+     */
+    private function getStatsByLeadAndType(int $leadId, string $type, array $options = []): array
+    {
+        $q = $this->getEntityManager()->getConnection()->createQueryBuilder();
+
+        $q->from(MAUTIC_TABLE_PREFIX.'focus_stats', 's')
+            ->select('s.id, s.lead_id, s.type, s.date_added, f.id as focus_id, f.name as focus_name')
+            ->leftJoin('s', MAUTIC_TABLE_PREFIX.'focus', 'f', 's.focus_id=f.id');
+
+        $q->where($q->expr()->andX(
+            $q->expr()->eq('s.lead_id', (int) $leadId),
+            $q->expr()->like('s.type', '"'.$type.'"')
+        ));
+
+        if (isset($options['search']) && $options['search']) {
+            $q->andWhere($q->expr()->orX(
+                $q->expr()->like('f.name', $q->expr()->literal('%'.$options['search'].'%')),
+                $q->expr()->like('f.description', $q->expr()->literal('%'.$options['search'].'%')),
+                $q->expr()->like('s.type', $q->expr()->literal('%'.$options['search'].'%'))
```

Comment:
Be careful with queries like this as searching with wildcards from the beginning is expensive as it cannot use indexes. So if the table grows into hundreds of thousands then searching will become unusable as it will time out the request and cause issues for the users. In many places we use only wildcard at the end of the string.

Also, does it make sense to search in the type? It's already handled above. But why the "like" operator was used there?

Reply:
Hi @escopecz, 
In the timeline search, page hits are also searched in this way. 
```
        // app/bundles/PageBundle/Entity/HitRepository.php  
        // Line: 86
        $query->andWhere($query->expr()->like('p.title', $query->expr()->literal('%'.$options['search'].'%')));        
```
e.g. 

without a wildcard from the beginning, there will be no focus named "[DPMMA-1743] focus test" in the result:
![image](https://user-images.githubusercontent.com/100592709/229822137-928d84ea-4866-4a2f-a202-a51891fa20ca.png)

ant with a wildcard from the beginning yes: 
![image](https://user-images.githubusercontent.com/100592709/229823162-5fb19f38-0c58-49f0-bd4f-731c97cab215.png)

A type search makes sense when its search terms "click" or "view":
![image](https://user-images.githubusercontent.com/100592709/229817344-e3c8a66e-299d-4f71-9690-8a4775792c14.png)

but It doesn't sense with "like" operator. I Changed this. 

In conclusion you are right that searching with wildcard is expensive. However, removing the wildcard from the beginning will limit the search possibilities. I leave the final decision in this matter to you :).